### PR TITLE
Replace many std::endl with "\n" sometimes with std::flush

### PR DESCRIPTION
**Pull request by pirapira**
Spin off from this discussion https://github.com/ethereum/cpp-ethereum/pull/4340#discussion_r133702673

### CODE REVIEW

**Comment by winsvega** [comment]
please paste the output of testeth to see how it affects the output

File: test/tools/fuzzTesting/createRandomTest.cpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ int createRandomTest(std::vector<char*> const& _parameters)
 
 	if (testSuite == "")
 	{
-		std::cerr << "Error! Test suite not supported! (Usage -t TestSuite)" << std::endl;
+		std::cerr << "Error! Test suite not supported! (Usage -t TestSuite)\n";
```
Comment (winsvega) [inline test/tools/fuzzTesting/createRandomTest.cpp]: maybe replace std::endl with  some marco or function/ custom operator?

File: test/tools/libtesteth/ImportTest.cpp [hunk 2]
```diff
@@ -642,7 +642,7 @@ void ImportTest::checkGeneralTestSectionSearch(json_spirit::mObject const& _expe
 				int errcode = compareStates(expectState, postState, stateMap, WhenError::Throw);
 				if (errcode > 0)
 				{
-					cerr << trInfo << std::endl;
+					cerr << trInfo << "\n" << std::flush;
```
Comment (chfast) [inline test/tools/libtesteth/ImportTest.cpp]: Why flush? cerr is not buffered so flush is never needed (am I wrong?). Flush is only needed in some rare cases in cout.

File: test/unittests/libdevcrypto/MemTrie.cpp [hunk 3]
```diff
@@ -49,7 +49,7 @@ class MemTrieNode
 	void putRLP(RLPStream& _parentStream) const;
 
 #if ENABLE_DEBUG_PRINT
-	void debugPrint(std::string const& _indent = "") const { std::cerr << std::hex << hash256() << ":" << std::dec << std::endl; debugPrintBody(_indent); }
+	void debugPrint(std::string const& _indent = "") const { std::cerr << std::hex << hash256() << ":" << std::dec << "\n" << std::flusyh; debugPrintBody(_indent); }
```
Comment (chfast) [inline test/unittests/libdevcrypto/MemTrie.cpp]: I'm guessing this is not compiled.

Comment (pirapira) [inline test/unittests/libdevcrypto/MemTrie.cpp]: Oh, me too.

File: test/tools/libtesteth/ImportTest.cpp (hunk 2 above)
Comment (pirapira) [inline test/tools/libtesteth/ImportTest.cpp]: Didn't know about that `cerr` / `cout` difference.

Comment (pirapira) [inline test/tools/libtesteth/ImportTest.cpp]: Fixed this and something similar in the same file.

File: test/unittests/libdevcrypto/MemTrie.cpp (hunk 3 above)
Comment (pirapira) [inline test/unittests/libdevcrypto/MemTrie.cpp]: Fixed the spelling.

Comment (pirapira) [inline test/unittests/libdevcrypto/MemTrie.cpp]: Ah, but this one is on cerr too.

Comment (pirapira) [inline test/unittests/libdevcrypto/MemTrie.cpp]: Removed the flush altogether.

**Comment by chfast** [comment]
I meant to remove all "flushes". I think the only case when explicit flush is needed **on std::cout** is when we need to use `std::system`.

**Comment by pirapira** [comment]
OK, removed `flush` except one.  The one is a spinner.  I want to see the `.` every ten seconds.

**Pull request merged**